--- qemu-6.2.0/linux-user/syscall.c	2021-12-14 15:42:02.000000000 -0500
+++ /home/jacob/qemu-test/qemu-6.2.0/linux-user/syscall.c	2022-03-18 19:04:07.498749898 -0400
@@ -140,6 +140,8 @@
 #include "fd-trans.h"
 #include "tcg/tcg.h"
 
+#include <sys/ptrace.h>
+
 #ifndef CLONE_IO
 #define CLONE_IO                0x80000000      /* Clone io context */
 #endif
@@ -8519,6 +8521,10 @@
         unlock_user(p, arg2, 0);
         return ret;
 #endif
+    case TARGET_NR_ptrace: {
+        ptrace(PTRACE_TRACEME, 0, 0, 0);
+        return 0;
+    }
     case TARGET_NR_execve:
         {
             char **argp, **envp;
